src/urllib3/connection.py

```diff
@@ -3,23 +3,42 @@
 import os
 import re
 import socket
+import sys
 import warnings
+from copy import copy
 from http.client import HTTPConnection as _HTTPConnection
 from http.client import HTTPException  # noqa: F401
 from socket import timeout as SocketTimeout
+from typing import (
+    IO,
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    Mapping,
+    NamedTuple,
+    Optional,
+    Tuple,
+    Union,
+    cast,
+)
 
 from .util.proxy import create_proxy_ssl_context
 from .util.util import to_str
 
-try:  # Compiled with SSL?
+if TYPE_CHECKING:
```

Comment:
Isn't `BaseSSLError` required for being imported elsewhere? If so we can't have `TYPE_CHECKING` bounding it

Reply:
Right, it is required by `connectionpool` and `response`. I assumed that type checking won't be performed in environments where there is no SSL, in that case should I just ignore everything in the `except` branch? Also, there are some issues regarding conditional imports in mypy https://github.com/python/mypy/issues/1153